### PR TITLE
Add flag to rnp_import_keys() which allows iterative key import.

### LINKED ISSUE #1273: Better strategy for importing a big public key block?
Although RNP has recently improved the memory allocation situation, it still requires a lot of RAM when processing a big public key block.

We see that users try to export all their public keys from GnuPG, and then try to import them to Thunderbird.

Ideally, it would be great to have a way to "split" the big key block that the user has exported, into smaller pieces, or even into individual keys, and then ask RNP to process each key individually.

Can you think of an easy way to perform this split?

### PR DESCRIPTION
**Pull request by ni4**
This PR:
- adds flag RNP_LOAD_SAVE_SINGLE, which will just load the first key from rnp_input_t
- makes use of this flag in CLI for better user experience (so progress is displayed during key load)
- adds corresponding tests.

Fixes #1273

### CODE REVIEW

**Review by antonsviridenko — APPROVED** [review]

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]
Please have a look at my remarks

File: src/librekey/key_store_pgp.cpp [hunk 1]
```diff
@@ -240,14 +240,42 @@ rnp_key_from_transferable_subkey(pgp_key_t *                subkey,
...
+{
+    pgp_transferable_key_t key;
+    rnp_result_t           ret = process_pgp_key_auto(src, key, true, skiperrors);
+
+    if (ret && (!skiperrors || (ret != RNP_ERROR_BAD_FORMAT))) {
+        return ret;
+    }
+
+    /* check whether we have primary key */
+    if (key.key.tag != PGP_PKT_RESERVED) {
```
[the remark this answers is not in the document]
Comment (ni4) [reply src/librekey/key_store_pgp.cpp:255]: This approach could give us a problem in (rather theoretical) case when in future 0 packet tag will get some meaning in the protocol, and we'll be forced to change default key packet tag value to something else.

File: src/librekey/key_store_pgp.h [hunk 2]
```diff
@@ -59,7 +59,13 @@
 #include <librepgp/stream-common.h>
 #include <librepgp/stream-key.h>
 
-rnp_result_t rnp_key_store_pgp_read_from_src(rnp_key_store_t *keyring, pgp_source_t *src);
+rnp_result_t rnp_key_store_pgp_read_from_src(rnp_key_store_t *keyring,
```
[the remark this answers is not in the document]
Comment (ni4) [reply src/librekey/key_store_pgp.h:62]: Ok, sure.

File: src/librepgp/stream-key.cpp [hunk 3]
```diff
@@ -680,6 +683,47 @@ process_pgp_subkey(pgp_source_t &src, pgp_transferable_subkey_t &subkey, bool sk
...
+                     bool                    skiperrors)
+{
+    key = {};
+    int ptag = stream_pkt_type(&src);
+    if (is_subkey_pkt(ptag) && allowsub) {
+        pgp_transferable_subkey_t subkey;
+        rnp_result_t              ret = process_pgp_subkey(src, subkey, skiperrors);
+        if (subkey.subkey.tag != PGP_PKT_RESERVED) {
+            try {
+                key.subkeys.emplace_back(std::move(subkey));
```
[the remark this answers is not in the document]
Comment (ni4) [reply src/librepgp/stream-key.cpp:699]: Agree, thanks. Wrote that code before the discussion took place.

**Comment by ni4** [comment]
@rrrooommmaaa Fixed all except checking for zero tag, let's discuss it further here: https://github.com/rnpgp/rnp/pull/1317#discussion_r495893588

**Review by rrrooommmaaa — APPROVED** [review]
LGTM, feel free to merge after tests are successful!

**Comment by ni4** [comment]
@rrrooommmaaa Thanks, all green - merging this.

**Pull request merged**